segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -0,0 +1,185 @@
+     * 2. If the segment is absent in the container metadata, then it is registered using the properties from the given
+     * {@link SegmentProperties}.
+     * @param container         A {@link DebugStreamSegmentContainer} instance for registering the given segment and checking
+     *                          its existence in the container metadata.
+     * @param storageSegment    A {@link SegmentProperties} instance which has properties of the segment to be registered.
+     * @return                  CompletableFuture which when completed will have the segment registered on to the container
+     *                          metadata.
+     */
+    private static CompletableFuture<Void> recoverSegment(DebugStreamSegmentContainer container, SegmentProperties storageSegment) {
+        Preconditions.checkNotNull(container);
+        Preconditions.checkNotNull(storageSegment);
+        long segmentLength = storageSegment.getLength();
+        boolean isSealed = storageSegment.isSealed();
+        String segmentName = storageSegment.getName();
+
+        log.info("Registering: {}, {}, {}.", segmentName, segmentLength, isSealed);
+        return Futures.exceptionallyComposeExpecting(
+                container.getStreamSegmentInfo(storageSegment.getName(), TIMEOUT)
+                        .thenAccept(e -> {
+                            if (segmentLength != e.getLength() || isSealed != e.isSealed()) {
+                                container.metadataStore.deleteSegment(segmentName, TIMEOUT)
+                                        .thenAccept(x -> container.registerSegment(segmentName, segmentLength, isSealed));
+                            }
+                        }), ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException,
+                () -> container.registerSegment(segmentName, segmentLength, isSealed));
+    }
+
+    /**
+     * Deletes container metadata segment and its Attribute segment from the {@link Storage} for the given container Id.
+     * @param storage       A {@link Storage} instance to delete the segments from.
+     * @param containerId   Id of the container for which the segments has to be deleted.
+     */
+    public static void deleteContainerMetadataAndAttributeSegments(Storage storage, int containerId) {
+        Preconditions.checkNotNull(storage);
+        String metadataSegmentName = NameUtils.getMetadataSegmentName(containerId);
+        String attributeSegmentName = NameUtils.getAttributeSegmentName(metadataSegmentName);
+        deleteSegment(storage, metadataSegmentName);
+        deleteSegment(storage, attributeSegmentName);
+    }
+
+    /**
+     * Deletes the segment with given name from the given {@link Storage} instance.
+     * @param storage       A {@link Storage} instance to delete the segments from.
+     * @param segmentName   Name of the segment to be deleted.
+     */
+    private static void deleteSegment(Storage storage, String segmentName) {
```

Comment:
This method should return CompletableFuture.

Reply:
Done.